src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -108,6 +108,31 @@
 }
 
 
+def all_services() -> List[Dict]:
+    orch = OrchClient.instance()
+    services = [service.to_dict() for service in orch.services.list()]
+    return services
+
+
+def hostnames_per_service(service_name: str) -> List[str]:
+    orch = OrchClient.instance()
+    daemons = orch.services.list_daemons(service_name=service_name)
+    daemons = [d.to_dict() for d in daemons]
+    result = [daemon['hostname'] for daemon in daemons]
+    return list(set(result))
+
+
+def hosts_per_service() -> List[Dict]:
+    hosts_count_per_service = []
+    services = all_services()
+    for svc in services:
+        daemons = hostnames_per_service(svc['service_name'])
+        hosts_count_per_service.append({'service': svc['service_name'],
+                                       'hosts': len(daemons)})
+
+    return hosts_count_per_service
```

Comment:
We can't use services.list or services.list_daemons as it is, we need to use `to_dict` to iterate them and `set` to avoid counting duplicate hostnames for each service

Reply:
I get the set() thing (actually the `{ }` is a set), what I don't get is the need of converting to dict? Isn't `list_daemons` returning a list of objects with a `.hostname` property?